designdocs/provider_contexts.md

```diff
@@ -0,0 +1,272 @@
+# Provider Contexts Design Doc
+
+## Context and Scope
+
+On Frequency, a Provider may either represent an application or a company.
+This document will outline the changes that will allow a Provider to represent a company with one or multiple applications.
+It also enables social wallet providers, custodial and non-custodial, to provide safe application information to their users.
+
+This is an expansion of the [Registered Provider](https://github.com/frequency-chain/frequency/blob/main/designdocs/provider_registration.md) concepts.
+
+Reference Document: [SIWF Specification](https://projectlibertylabs.github.io/siwf/v2/docs/)
+
+### Why Is This Needed?
+
+#### Concept of an Application Separate from a Provider
+
+Many companies have a single major product, but some companies have multiple applications that serve different needs.
+While each application could be a separate provider, this causes two issues: The sharing of capacity and the user permissions.
+
+Having separate providers would require a single company to have separate Capacity staking for each application.
+As each these applications are part of the same company, those applications should share Capacity as a single legal entity.
+
+Separate providers per application would NOT align with the data security model of user-to-company that exists in the world today.
+Provider-level delegations are better than introducing application-level delegations.
```

Comment:
Can you please help me understand why Provider-level delegations are better than application-level delegations?

Reply:
```suggestion
For example, imagine if Alice is delegating to an application, Bar. Bar application is owned by the Foo Company which also owns and controls Baz. The Foo Company is able to collect all data shared with both the Bar and the Baz. A delegation to just Bar or just Baz doesn't limit the information sharing between those Applications because the data agreement is with Foo Company.
Additionally, to limit the impact on the network, the Bar and Baz Applications are able to share capacity and perform batching together.
This leads to the conclusion that Provider-level delegations are better than introducing new Application-level delegations.
```

Reply:
@enddynayn how does that work?

Reply:
Perfect! Thanks!